### PR TITLE
BB Internal audit fixes

### PR DESCRIPTION
**Pull request by aalavandhan**
Should cover the changes discussed in the google doc.

### CODE REVIEW

File: spot-vaults/contracts/_strategies/SpotAppraiser.sol [hunk 1]
```diff
@@ -49,29 +46,34 @@ contract SpotAppraiser is Ownable, IBillBrokerPricingStrategy {
...
+    uint256 private constant SPOT_DR_ONE = (10 ** SPOT_DR_DECIMALS);
     uint256 public constant CL_ORACLE_DECIMALS = 8;
-    uint256 public constant CL_ORACLE_STALENESS_TRESHOLD_SEC = 3600 * 24; // 1 day
+    uint256 public constant CL_ORACLE_STALENESS_THRESHOLD_SEC = 3600 * 48; // 2 days
     uint256 public constant USD_LOWER_BOUND = (99 * ONE) / 100; // 0.99$
-    uint256 public constant AMPL_DUST_AMT = 1000 * (10 ** 9); // 1000 AMPL
-
-    /// @notice Address of the AMPL ERC-20 token contract.
-    IAMPL public immutable AMPL;
+    uint256 public constant AMPL_DUST_AMT = 25000 * (10 ** 9); // 25000 AMPL
```
Comment (aalavandhan) [inline spot-vaults/contracts/_strategies/SpotAppraiser.sol:54]: Increased this to 25,000 AMPL (above the discussed 10k), to match with the rollover vault's min-deployment-amount. Which means if someone griefs by sending 25k ampl into perp, the vault should immediately roll it over.

File: spot-vaults/contracts/BillBroker.sol [hunk 2]
```diff
@@ -748,27 +738,33 @@ contract BillBroker is
...
-        // Area between [xL, xU] => (m . (xU^2 - xL^2) / 2 + c . (xU - xL)) / (xU - xL)
-        //                       => m.(xL+xU)/2 + c
+        // Avg height => (yL + yU) / 2
+        //            => m . ( xL + xU ) / 2 + c
         int256 dlY = fn.y2.toInt256() - fn.y1.toInt256();
         int256 dlX = fn.x2.toInt256() - fn.x1.toInt256();
         int256 c = fn.y2.toInt256() - ((fn.x2.toInt256() * dlY) / dlX);
-        int256 area = ((xL + xU).toInt256() * dlY) / (2 * dlX) + c;
-        return area.abs();
+        return ((((xL + xU).toInt256() * dlY) / (2 * dlX)) + c).abs();
```
Comment (aalavandhan) [inline spot-vaults/contracts/BillBroker.sol:768]: Though the final avg height is always positive, intermediate values can be negative (the slope of the line, y intercept etc). So Im leaning toward keeping the math here as is (ie casting to signed int and casting back).

File: spot-vaults/contracts/BillBroker.sol [hunk 3]
```diff
@@ -657,7 +641,12 @@ contract BillBroker is
                     x2: arSoftBound.lower,
                     y2: swapFeePercs.lower
                 }),
-                Line({ x1: 0, y1: swapFeePercs.lower, x2: ONE, y2: swapFeePercs.lower }),
+                Line({
+                    x1: arSoftBound.lower,
+                    y1: swapFeePercs.lower,
+                    x2: ONE,
```
Comment (brandoniles) [inline spot-vaults/contracts/BillBroker.sol:647]: Should this be an ~infinite value?

Comment (brandoniles) [reply spot-vaults/contracts/BillBroker.sol:647]: I have a question about the Line inputs to the other _computeFeePerc call too, but maybe easiest to sync offline.

File: spot-vaults/contracts/BillBroker.sol [hunk 4]
```diff
@@ -313,12 +296,11 @@ contract BillBroker is
         whenNotPaused
         returns (uint256 usdAmtOut, uint256 perpAmtOut)
     {
-        if (burnAmt <= 0) {
+        (usdAmtOut, perpAmtOut) = computeRedemptionAmts(burnAmt);
+        if (usdAmtOut == 0 || perpAmtOut == 0) {
```
[the remark this answers is not in the document]
Comment (aalavandhan) [reply spot-vaults/contracts/BillBroker.sol:300]: Yea good catch. I think we need `usdAmtOut == 0 && perpAmtOut == 0`

File: spot-vaults/contracts/_strategies/SpotAppraiser.sol [hunk 5]
```diff
@@ -195,12 +161,11 @@ contract SpotAppraiser is Ownable, IBillBrokerPricingStrategy {
...
 
     /// @return If the spot token is healthy.
-    function isSPOTHealthy() public view returns (bool) {
+    function isSPOTHealthy() public returns (bool) {
         // If the SPOT's `deviationRatio` is lower than the defined bound
         // i.e) it doesn't have enough capital to cover future rollovers,
         // we consider it unhealthy.
-        IBalancer balancer = SPOT.balancer();
-        uint256 spotDR = balancer.deviationRatio().mulDiv(ONE, 10 ** balancer.decimals());
+        uint256 spotDR = SPOT.deviationRatio().mulDiv(ONE, SPOT_DR_ONE);
```
[the remark this answers is not in the document]
Comment (aalavandhan) [reply spot-vaults/contracts/_strategies/SpotAppraiser.sol:168]: Right

File: spot-vaults/contracts/BillBroker.sol [hunk 6]
```diff
@@ -493,18 +469,19 @@ contract BillBroker is
...
-            perpAmtIn = perpReserve_.mulDiv(usdAmtIn, usdReserve_);
-            mintAmt = totalSupply_.mulDiv(usdAmtIn, usdReserve_);
+            perpAmtIn = perpBalance_.mulDiv(usdAmtIn, usdBalance_);
             if (perpAmtIn > perpAmtMax) {
                 perpAmtIn = perpAmtMax;
-                usdAmtIn = usdReserve_.mulDiv(perpAmtIn, perpReserve_);
-                mintAmt = totalSupply_.mulDiv(perpAmtIn, perpReserve_);
+                usdAmtIn = usdBalance_.mulDiv(perpAmtIn, perpBalance_);
             }
+            mintAmt = totalSupply_.mulDiv(usdAmtIn, usdBalance_);
```
[the remark this answers is not in the document]
Comment (aalavandhan) [reply spot-vaults/contracts/BillBroker.sol:483]: hmm. Not sure if we want to allow `100:0` and `0:100` as they will make the asset ratio (or other intermediate values) will become undefined

File: spot-vaults/contracts/_strategies/SpotAppraiser.sol (hunk 5 above)
[the remark this answers is not in the document]
Comment (brandoniles) [reply spot-vaults/contracts/_strategies/SpotAppraiser.sol:168]: Pushing to a potential followup

**Review by brandoniles — APPROVED** [review]
LGTM

**Pull request merged**